### PR TITLE
feat(code splitting): add code splitting with react-loadable

### PR DESCRIPTION
**Pull request by Yangzhedi**
#7540 

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by deepu105** [comment]
Thanks I'll check it out when I have some time

**Comment by deepu105** [comment]
Its ok don't worry, if the approach is acceptable the PR would need a
change anyway

Thanks & Regards,
Deepu

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -37,6 +37,7 @@ limitations under the License.
     "react-addons-transition-group": "16.0.0-alpha.3",
     "react-dom": "16.3.0",
     "react-hot-loader": "3.1.1",
+    "react-loadable": "^5.3.1",
```
Comment (pascalgrimaud) [inline generators/client/templates/react/package.json.ejs]: can you remove the `^` and use a fixed version plz ?

**Comment by Yangzhedi** [comment]
@deepu105   that`s ok   thank you for your reply.

**Comment by Yangzhedi** [comment]
@deepu105   that`s ok   thank you for your reply.

File: generators/client/templates/react/src/main/webapp/app/routes.tsx.ejs [hunk 2]
```diff
@@ -34,6 +34,17 @@ import Entities from 'app/entities';
 import PrivateRoute from 'app/shared/auth/private-route';
 import { AUTHORITIES } from 'app/config/constants';
 
+<%_ if (authenticationType !== 'oauth2') { _%>
+/*
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/routes.tsx.ejs]: Remove the comments here, instead add some docs on the generated Readme file about this

Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/routes.tsx.ejs]: In Angular, we have lazy loading for admin modules, so can you do this for all the admin module components as well?

Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/routes.tsx.ejs]: also please test with prod profile

**Comment by deepu105** [comment]
@Yangzhedi can you please check why the react builds are failing in travis?

**Comment by Yangzhedi** [comment]
@deepu105 sorry, i have no idea about the error in travis.

```
Failed to parse file: /home/travis/app/src/main/webapp/app/config/notification-middleware.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-dto.model.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-pagination-and-dto.model.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-pagination.model.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-service-class-and-dto.model.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-service-class-and-pagination.model.ts
Failed to parse file: /home/travis/app/src/main/webapp/app/shared/model/entity-with-service-class-pagination-and-dto.model.ts
...
```

**Comment by deepu105** [comment]
@chrisdns @agaspardcilia does any of you have bandwidth to check the issue here?

**Comment by deepu105** [comment]
I guess I fixed the issues.

@jhipster/developers check this. This is how easy lazy loading is in React :wink:

**Pull request merged**